pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -0,0 +1,88 @@
+package component
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/golang/glog"
+
+	adapterCommon "github.com/openshift/odo/pkg/devfile/adapters/common"
+	"github.com/openshift/odo/pkg/devfile/versions/common"
+	"github.com/openshift/odo/pkg/kclient"
+)
+
+// New instantiantes a component adapter
+func New(commonAdapter adapterCommon.AdapterContext, client kclient.Client) Adapter {
+	return Adapter{
+		Client:         client,
+		AdapterContext: commonAdapter,
+	}
+}
+
+// Adapter is a component adapter implementation for Kubernetes
+type Adapter struct {
+	Client kclient.Client
+	adapterCommon.AdapterContext
+}
+
+// GetContainers iterates through the components in the devfile and returns a slice of the corresponding containers
+func (a *Adapter) GetContainers() []corev1.Container {
+	var containers []corev1.Container
+	// Only components with aliases are considered because without an alias commands cannot reference them
+	for _, comp := range a.Devfile.Data.GetAliasedComponents() {
+		if comp.Type == common.DevfileComponentTypeDockerimage {
+			glog.V(3).Infof("Found component %v with alias %v\n", comp.Type, *comp.Alias)
+			envVars := convertEnvs(comp.Env)
+			resourceReqs := getResourceReqs(comp)
+			container := kclient.GenerateContainer(*comp.Alias, *comp.Image, false, comp.Command, comp.Args, envVars, resourceReqs)
+			containers = append(containers, *container)
+		}
+	}
+	return containers
+}
+
+// Start updates the component if a matching component exists or creates one if it doesn't exist
+func (a Adapter) Start() (err error) {
+	componentName := a.ComponentName
+
+	labels := map[string]string{
+		"component": componentName,
+	}
+
+	containers := a.GetContainers()
+	if len(containers) == 0 {
+		return fmt.Errorf("No valid components found in the devfile")
+	}
+
+	objectMeta := kclient.CreateObjectMeta(componentName, a.Client.Namespace, labels, nil)
+	podTemplateSpec := kclient.GeneratePodTemplateSpec(objectMeta, a.Client.Namespace, containers)
```

Comment:
`GeneratePodTemplateSpec()` second argument is the service account and not the namespace, even though `GeneratePodTemplateSpec()` is not using it with this PR, pls see comment on that func.

Reply:
I think I know what happened here. It looks like you removed namespace from your PR and I had removed namespace from mine and when I rebased on the changes from master I got my change on this line and yours on the other line. I'll update the code to use namespace from the `objectMeta` and just remove service account for now as mentioned in the comment above.